src/main/java/com/aws/greengrass/clientdevices/auth/iot/IotAuthClient.java

```diff
@@ -122,12 +145,58 @@ public boolean isThingAttachedToCertificate(Thing thing, Certificate certificate
-                                + " grants the greengrass:VerifyClientDeviceIoTCertificateAssociation permission.");
+                        .log("Failed to verify certificate thing association. Check that the core device's "
+                                + "IoT policy grants the greengrass:VerifyClientDeviceIoTCertificateAssociation "
+                                + "permission.");
                 throw new CloudServiceInteractionException(
                         String.format("Failed to verify certificate %s thing %s association",
                                 certificate.getCertificateId(), thing.getThingName()), e);
             }
         }
+
+        @Override
+        public Stream<Thing> getThingsAssociatedWithCoreDevice() {
+            DeviceConfiguration configuration = clientFactory.getDeviceConfiguration();
+            String thingName = Coerce.toString(configuration.getThingName());
+
+            ListClientDevicesAssociatedWithCoreDeviceRequest request =
+                    ListClientDevicesAssociatedWithCoreDeviceRequest.builder()
+                            .coreDeviceThingName(thingName)
+                            .build();
+
+            try (GreengrassV2Client client = getGGV2Client(configuration)) {
+                ListClientDevicesAssociatedWithCoreDeviceResponse response =
+                        client.listClientDevicesAssociatedWithCoreDevice(request);
+
+                return response.associatedClientDevices().stream()
+                        .map(AssociatedClientDevice::thingName)
+                        .map(Thing::of);
+            }  catch (SdkClientException | AwsServiceException e) {
+                throw new CloudServiceInteractionException("Failed to list things associated with core", e);
+            }
+        }
+
+        // TODO: This should not live here ideally it should be returned by the clientFactory but we
+        //  are adding it here to avoid introducing new changes to the nucleus
+        private GreengrassV2Client getGGV2Client(DeviceConfiguration deviceConfiguration) {
+            ApacheHttpClient.Builder httpClient = ClientConfigurationUtils
+                    .getConfiguredClientBuilder(deviceConfiguration);
+
+            String awsRegion = Coerce.toString(deviceConfiguration.getAWSRegion());
+            GreengrassV2ClientBuilder clientBuilder = GreengrassV2Client.builder()
+                    .httpClient(httpClient.build())
+                    .credentialsProvider(lazyCredentialProvider)
+                    .overrideConfiguration(
+                            ClientOverrideConfiguration.builder().retryPolicy(RetryMode.STANDARD).build());
+
+            if (!Utils.isEmpty(awsRegion)) {
+                String greengrassServiceEndpoint = RegionUtils.getGreengrassControlPlaneEndpoint(
+                        awsRegion,  IotSdkClientFactory.EnvironmentStage.PROD);
```

Comment:
don't hardcode the environment, this must pull the environment from the device config.